.github/workflows/releaseArtifacts.yml

```diff
@@ -48,38 +50,60 @@ jobs:
-        uses: actions/upload-artifact@6f51ac03b9356f520e9adb1b1b7802705f340c2b # v4.5.0
-        with:
-          path: ./build/launch4j/Stirling-PDF-Server${{ matrix.file_suffix }}.exe
-          name: Stirling-PDF-Server${{ matrix.file_suffix }}.exe
-          overwrite: true
-          retention-days: 1
-          if-no-files-found: error
+      - name: Install Cosign
+        uses: sigstore/cosign-installer@dc72c7d5c4d10cd6bcb8cf6e3fd625a9e5e537da # v3.7.0
 
-      - name: Upload binaries to release
-        uses: softprops/action-gh-release@01570a1f39cb168c169c802c3bceb9e93fb10974 # v2.1.0
-        with:
-          files: ./build/launch4j/Stirling-PDF-Server${{ matrix.file_suffix }}.exe
+      - name: Sign JAR with Cosign OIDC
+        run: |
+          cosign sign-blob \
+            --yes \
+            --oidc-client-id sigstore \
+            --oidc-issuer https://token.actions.githubusercontent.com \
+            ./build/libs/Stirling-PDF${{ matrix.file_suffix }}.jar \
+            > ./build/libs/Stirling-PDF${{ matrix.file_suffix }}.jar.sig
 
-      - name: Rename jar binaries
-        run: cp ./build/libs/Stirling-PDF-${{ steps.versionNumber.outputs.versionNumber }}.jar ./build/libs/Stirling-PDF${{ matrix.file_suffix }}.jar
+      - name: Sign EXE with Cosign OIDC
+        run: |
+          cosign sign-blob  \
+            --yes \
+            --oidc-client-id sigstore \
+            --oidc-issuer https://token.actions.githubusercontent.com \
+            ./build/launch4j/Stirling-PDF-Server${{ matrix.file_suffix }}.exe \
+            > ./build/launch4j/Stirling-PDF-Server${{ matrix.file_suffix }}.exe.sig
 
-      - name: Upload Assets jar binaries
-        uses: actions/upload-artifact@6f51ac03b9356f520e9adb1b1b7802705f340c2b # v4.5.0
-        with:
-          path: ./build/libs/Stirling-PDF${{ matrix.file_suffix }}.jar
-          name: Stirling-PDF${{ matrix.file_suffix }}.jar
-          overwrite: true
-          retention-days: 1
-          if-no-files-found: error
+      - name: Generate Attestations for JAR
+        run: |
+          cosign attest-blob  \
+            --predicate - \
+            --yes \
+            --oidc-client-id sigstore \
+            --oidc-issuer https://token.actions.githubusercontent.com \
+            ./build/libs/Stirling-PDF${{ matrix.file_suffix }}.jar \
+            > ./build/libs/Stirling-PDF${{ matrix.file_suffix }}.jar.intoto.jsonl
+
+      - name: Generate Attestations for EXE
+        run: |
+          cosign attest-blob  \
+            --predicate - \
+            --yes \
+            --oidc-client-id sigstore \
+            --oidc-issuer https://token.actions.githubusercontent.com \
+            ./build/launch4j/Stirling-PDF-Server${{ matrix.file_suffix }}.exe \
+            > ./build/launch4j/Stirling-PDF-Server${{ matrix.file_suffix }}.exe.intoto.jsonl
 
-      - name: Upload jar binaries to release
+      - name: Upload binaries, attestations and signatures to release
```

Comment:
I would avoid using an external action to upload binaries. Gh CLI is already available inside GH runner, you can use something like (haven't tested it but should be close to that)
https://cli.github.com/manual/gh_release_upload
```yaml
 - name: Upload artifact to release
   run: gh release upload ${{ github.ref }} ./build/libs/Stirling-PDF${{ matrix.file_suffix }}.* ./build/launch4j/Stirling-PDF-Server${{ matrix.file_suffix }}.*
   env:
     GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```